torch/csrc/autograd/engine.cpp

```diff
@@ -630,6 +631,9 @@ void validate_outputs(
       ss << metadata.device() << " but got " << grad_device;
       AT_ERROR(format_error(ss.str()));
     }
+    // The gradient type should match the forward Tensor one
```

Comment:
not sure what the first line is in reference to, it's not checked here.

Reply:
Yes, I will remove this.